contributing/docs-style.md

```diff
@@ -0,0 +1,379 @@
+    <th style="vertical-align : middle;"><center><b>Possible Values</b></center></th>
+    <th style="vertical-align : middle;"><center><b>Description</b></center></th>
+  </tr>
+  <tr>
+    <td><b>Layout</b></td>
+    <td>layout</td>
+    <td>{base,plain,redirect}</td>
+    <td>The layout file to use. Layout files are located under the <i>_layouts</i> directory.</td>
+  </tr>
+  <tr>
+    <td><b>Content</b></td>
+    <td>title</td>
+    <td>%s</td>
+    <td>The title to be used as the top-level (Level-1) heading for the page.</td>
+  </tr>
+ </thead>
+</table>
+</font>
+
+Documentation-wide information and configuration settings that sit under `_config.yml` are also available to the front matter through the site variable. These settings can be accessed using the following syntax:
+
+```liquid
+{{ "{{ site.CONFIG_KEY " }}}}
+```
+
+The placeholder will be replaced with the value of the variable named `CONFIG_KEY` when generating the documentation.
+
+[Back to top](#top)
+
+## Formatting
+
+Listed in the following sections are the basic formatting guidelines to get you started with writing documentation that is consistent and simple to navigate.
+
+### Headings
+
+In Markdown, headings are any line prefixed with a hash (#), with the number of hashes indicating the level of the heading. Headings should be nested and consecutive — never skip a header level for styling reasons!
+
+<font size="3">
+<table width="100%" class="table table-bordered">
+  <thead>
+  <tr>
+    <th style="vertical-align : middle;"><center><b>Syntax</b></center></th>
+    <th style="vertical-align : middle;"><center><b>Level</b></center></th>
+    <th style="vertical-align : middle;"><center><b>Description</b></center></th>
+  </tr>
+  <tr>
+    <td># Heading</td>
+    <td><center>Level-1</center></td>
+    <td>The page title is defined in the Front Matter, so this level should <b>not be used</b>.</td>
+  </tr>
+  <tr>
+    <td>## Heading</td>
+    <td><center>Level-2</center></td>
+    <td>Starting level for Sections. Used to organize content by higher-level topics or goals.</td>
+  </tr>
+  <tr>
+    <td>### Heading</td>
+    <td><center>Level-3</center></td>
+    <td rowspan="2" style="vertical-align : middle;">Sub-sections. Used in each Section to separate supporting information or tasks.</td>
+  </tr>
+  <tr>
+    <td>#### Heading</td>
+    <td><center>Level-4</center></td>
+  </tr>
+</thead>
+</table>
+</font>
+
+<div class="alert alert-warning">
+  <h3>Best Practice</h3>
+  Use descriptive language in the phrasing of headings. For example, for a documentation page on dynamic tables, "Dynamic Tables and Continuous Queries" is more descriptive than "Background" or "Technical Information".
+</div>
+
+### Table of Contents
+
+In the documentation build, the **Table Of Contents** (TOC) is automatically generated from the headings of the page using the following line of markup:
+
+```liquid
+{{ "{% toc " }}%}
```

Comment:
This is interesting, the Flink website uses `{% toc %}` while the Flink Documentation uses `{:toc}`, for the doc we should probably stick to `{:toc}` then.